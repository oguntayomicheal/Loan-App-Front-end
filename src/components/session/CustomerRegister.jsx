import React, { useState } from 'react';
import { Link } from 'react-router-dom';

import { FaSpinner } from 'react-icons/fa';
import '../../assets/styles/LoadingSpinner.css';
import '../../assets/styles/session.css';

const CustomerRegister = () => {
  const [formData, setFormData] = useState({
    name: '',
    email: '',
    password: '',
    password_confirmation: '',
  });

  const [registrationSuccess, setRegistrationSuccess] = useState(false);
  const [registrationFailure, setRegistrationFailure] = useState(false);
  const [errorData, setErrorData] = useState(null);

  const [loading, setLoading] = useState(false);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((prevData) => ({
      ...prevData,
      [name]: value,
    }));
  };

  const url = 'http://127.0.0.1:3100/api/v1/customers';

  const handleSubmit = async (e) => {
    e.preventDefault();
    setLoading(true);
    setRegistrationFailure(false);

    try {
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ customer: formData }),
      });

      if (response.ok) {
        setRegistrationSuccess(true);
        setLoading(false);
      } else {
        const data = await response.json();
        setErrorData(data.errors);
        setRegistrationFailure(true);
        setLoading(false);
      }
    } catch (error) {
      throw new Error(error);
    }
  };

  return (
    <div className="session_page main_content">
      <h2>Register</h2>
      {registrationSuccess ? (
        <div className="reg_success">
          <p>Registration successful! You can now log in.</p>
          <Link to="/CustomerLogin" className="splash_link">Login</Link>
        </div>
      ) : (
        <form onSubmit={handleSubmit}>
          <label htmlFor="name">
            Name:
            <br />
            <input
              id="name"
              type="text"
              name="name"
              value={formData.name}
              onChange={handleChange}
            />
          </label>
          {registrationFailure && errorData && errorData.name && (
          <p className="reg_error">
            Name
            {' '}
            {errorData.name[0]}
          </p>
          )}
          <br />
          <label htmlFor="email">
            Email:
            <br />
            <input
              id="email"
              type="email"
              name="email"
              value={formData.email}
              onChange={handleChange}
            />
          </label>
          {registrationFailure && errorData && errorData.email && (
          <p className="reg_error">
            Email
            {' '}
            {errorData.email[0]}
          </p>
          )}
          <br />
          <label htmlFor="password">
            Password:
            <br />
            <input
              id="password"
              type="password"
              name="password"
              value={formData.password}
              onChange={handleChange}
            />
          </label>

          {registrationFailure && errorData && errorData.password && (
          <p className="reg_error">
            Password
            {' '}
            {errorData.password[0]}
          </p>
          )}
          <br />
          <label htmlFor="confirm_password">
            Confirm Password:
            <br />
            <input
              id="confirm_password"
              type="password"
              name="password_confirmation"
              value={formData.password_confirmation}
              onChange={handleChange}
            />
          </label>
          {registrationFailure && errorData && errorData.password_confirmation && (
          <p className="reg_error">
            Password
            {' '}
            {errorData.password_confirmation[0]}
          </p>
          )}

          {loading && (

          <div className="loading-spinner">
            <FaSpinner className="icon" />
          </div>

          )}
          <br />
          <div className="reg_btns">
            <button type="submit" className="splash_link login_btn">Register</button>
            <Link to="/CustomerLogin" className="splash_link">Login</Link>
          </div>
        </form>
      )}

    </div>
  );
};

export default CustomerRegister;
