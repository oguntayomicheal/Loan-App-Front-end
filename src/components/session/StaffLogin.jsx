import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';

import { FaSpinner } from 'react-icons/fa';
import '../../assets/styles/LoadingSpinner.css';
import '../../assets/styles/session.css';

const StaffLogin = () => {
  const [staffID, setStaffID] = useState('');

  const [loading, setLoading] = useState(false);
  const [loginFailure, setLoginFailure] = useState(false);

  const navigate = useNavigate();

  const handleSubmit = async (e) => {
    e.preventDefault();
    setLoading(true);
    setLoginFailure(false);

    const url = 'http://127.0.0.1:3100/api/v1/staffs/login';

    try {
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ staff_id: staffID }),
      });

      if (response.ok) {
        const data = await response.json();
        localStorage.setItem('staffData', JSON.stringify(data));
        navigate('/StaffDashBoard');
      } else {
        setLoginFailure(true);
        setLoading(false);
      }
    } catch (error) {
      throw new Error(error);
    }
  };

  return (
    <div className='session_page main_content'>
      <h1>Staff Login Page</h1>

      <form onSubmit={handleSubmit}>
        <input
          type="password"
          id="staffID"
          name="staffID"
          placeholder="Enter your Staff ID"
          value={staffID}
          onChange={(event) => setStaffID(event.target.value)}
        />

        <button type="submit">
          Login
        </button>

      </form>

      {loading && (

      <div className="loading-spinner">
        <FaSpinner className="icon" />
      </div>

      )}

      {loginFailure && (
      <div>
        Invalid Staff ID
      </div>
      )}

    </div>
  );
};

export default StaffLogin;
