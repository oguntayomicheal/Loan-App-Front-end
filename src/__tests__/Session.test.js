import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import CustomerLogin from '../components/session/CustomerLogin';
import CustomerRegister from '../components/session/CustomerRegister';
import StaffLogin from '../components/session/StaffLogin';

describe('Session pages renders correctly', () => {
  test('for Customer Login page', () => {
    const customerLogin = render(
      <Router>
        <CustomerLogin />
      </Router>,
    );

    expect(customerLogin).toMatchSnapshot();
  });

  test('for Customer Registration page', () => {
    const customerRegistration = render(
      <Router>
        <CustomerRegister />
      </Router>,
    );

    expect(customerRegistration).toMatchSnapshot();
  });

  test('for Staff Login page', () => {
    const staffLogin = render(
      <Router>
        <StaffLogin />
      </Router>,
    );

    expect(staffLogin).toMatchSnapshot();
  });
});
