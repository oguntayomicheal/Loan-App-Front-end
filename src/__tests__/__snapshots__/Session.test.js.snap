// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Session pages renders correctly for Customer Login page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="session_page main_content"
      >
        <h2
          class="h2_login"
        >
          Login
        </h2>
        <form>
          <label
            for="email"
          >
            Email:
            <br />
            <input
              id="email"
              name="email"
              type="email"
              value=""
            />
          </label>
          <br />
          <label
            for="password"
          >
            Password:
            <br />
            <input
              id="password"
              name="password"
              type="password"
              value=""
            />
          </label>
          <br />
          <button
            class="splash_link login_btn"
            type="submit"
          >
            Login
          </button>
          <a
            class="splash_link"
            href="/CustomerRegister"
          >
            Register
          </a>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="session_page main_content"
    >
      <h2
        class="h2_login"
      >
        Login
      </h2>
      <form>
        <label
          for="email"
        >
          Email:
          <br />
          <input
            id="email"
            name="email"
            type="email"
            value=""
          />
        </label>
        <br />
        <label
          for="password"
        >
          Password:
          <br />
          <input
            id="password"
            name="password"
            type="password"
            value=""
          />
        </label>
        <br />
        <button
          class="splash_link login_btn"
          type="submit"
        >
          Login
        </button>
        <a
          class="splash_link"
          href="/CustomerRegister"
        >
          Register
        </a>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Session pages renders correctly for Customer Registration page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="session_page main_content"
      >
        <h2>
          Register
        </h2>
        <form>
          <label
            for="name"
          >
            Name:
            <br />
            <input
              id="name"
              name="name"
              type="text"
              value=""
            />
          </label>
          <br />
          <label
            for="email"
          >
            Email:
            <br />
            <input
              id="email"
              name="email"
              type="email"
              value=""
            />
          </label>
          <br />
          <label
            for="password"
          >
            Password:
            <br />
            <input
              id="password"
              name="password"
              type="password"
              value=""
            />
          </label>
          <br />
          <label
            for="confirm_password"
          >
            Confirm Password:
            <br />
            <input
              id="confirm_password"
              name="password_confirmation"
              type="password"
              value=""
            />
          </label>
          <br />
          <div
            class="reg_btns"
          >
            <button
              class="splash_link login_btn"
              type="submit"
            >
              Register
            </button>
            <a
              class="splash_link"
              href="/CustomerLogin"
            >
              Login
            </a>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="session_page main_content"
    >
      <h2>
        Register
      </h2>
      <form>
        <label
          for="name"
        >
          Name:
          <br />
          <input
            id="name"
            name="name"
            type="text"
            value=""
          />
        </label>
        <br />
        <label
          for="email"
        >
          Email:
          <br />
          <input
            id="email"
            name="email"
            type="email"
            value=""
          />
        </label>
        <br />
        <label
          for="password"
        >
          Password:
          <br />
          <input
            id="password"
            name="password"
            type="password"
            value=""
          />
        </label>
        <br />
        <label
          for="confirm_password"
        >
          Confirm Password:
          <br />
          <input
            id="confirm_password"
            name="password_confirmation"
            type="password"
            value=""
          />
        </label>
        <br />
        <div
          class="reg_btns"
        >
          <button
            class="splash_link login_btn"
            type="submit"
          >
            Register
          </button>
          <a
            class="splash_link"
            href="/CustomerLogin"
          >
            Login
          </a>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Session pages renders correctly for Staff Login page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="session_page main_content"
      >
        <h1>
          Staff Login Page
        </h1>
        <form>
          <input
            id="staffID"
            name="staffID"
            placeholder="Enter your Staff ID"
            type="password"
            value=""
          />
          <button
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="session_page main_content"
    >
      <h1>
        Staff Login Page
      </h1>
      <form>
        <input
          id="staffID"
          name="staffID"
          placeholder="Enter your Staff ID"
          type="password"
          value=""
        />
        <button
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
